packages/@glimmer/syntax/lib/errors/syntax-error.ts

```diff
@@ -0,0 +1,17 @@
+import * as AST from '../types/nodes';
+
+/*
+ * Subclass of `Error` with additional information
+ * about location of incorrect markup.
+ */
+class SyntaxError extends Error {
+  location: AST.SourceLocation;
```

Comment:
`loc` is kind of ambiguous tbh, I'd prefer `location` (also the property name that `broccoli-builder` expects is `location` not `loc`). If you feel strongly that we should stay consistent here, I can rename.